### ISSUE TITLE
add support for simulating timeout of a request

### ISSUE THREAD

**Issue by jhkuperus** [KrekkieD/nocca#21]

**Comment by KrekkieD** [comment]
How? failure percentage on endpoints / requests?

**Comment by jhkuperus** [comment]
That could be configured as a responseTransformmation.

I would also expect it to be useful in scenarios. Where you follow a path and want a certain request to timeout.

**Comment by qamil** [comment]
Is this feature planned already? We need it for our load testing setup.
Could you elaborate a bit on what needs to be done to implement this? Maybe we can collaborate on implementing it, depending on how much work it is.

**Comment by KrekkieD** [comment]
There is delay functionality included, which will delay a response from Nocca to the calling client. If your client has a timeout of 30 seconds, you could set a delay of 30+ seconds on the Nocca request causing a timeout in your app. Would that be an option?

**Comment by jkaizer** [comment]
We can use the delay option. However, as I understand it, it gives all responses the same delay. I would like to set a delay for each url or url pattern that is cached. Is that possible?

Our app talks to a collection of api's where one api has an average response of 1 second and another api  of 7 seconds.

**Comment by KrekkieD** [comment]
Do you use multiple endpoints in the nocca config? You can set the delay config for each endpoint.

**Comment by jkaizer** [comment]
Yes, but within one endpoint we also have multiple urls that we cache.


--
Jasper Kaizer
Q42  | tel +31 70 44523 42 | mob +31 6 39 65 52 76 |
http://q42.nl | @q42

**Comment by KrekkieD** [comment]
It may be better to split it up to multiple endpoints, but hard to day without knowing in more detail. Nonetheless the issue may be valid. 

Any ideas on how you would want to be able to configure this? I suppose those urls would need to be identifiable and groupable by meeting some condition, so the delay config could be added to any request on that endpoint that fulfills that condition. 

But what would be a logical way to set that up? RequestKey generation is all the same for each incoming request on a specific endpoint. This is also why it may be better to split the single endpoint into multiple endpoints.

Additional question; in your case, how do you (as a person) determine the delay value of an incoming request? Is it part of the url path? Or query params? Or something else?
